### PR TITLE
Adds styling, base content, and new template

### PR DESCRIPTION
**Pull request by barnesjoseph**

### CODE REVIEW

File: docs/source/_static/images/hero.svg
Comment (jakevdp) [inline docs/source/_static/images/hero.svg:1]: I think this was an early draft of the logo. Do you have an updated version?

File: docs/source/conf.py [hunk 1]
```diff
@@ -16,6 +16,11 @@
 
 extensions = [
     'myst_nb',
+    'sphinx.ext.duration',
+    'sphinx.ext.doctest',
+    'sphinx.ext.autodoc',
+    'sphinx.ext.autosummary',
```
Comment (jakevdp) [inline docs/source/conf.py:22]: I think lines 20-22 are unnecessary because we don't have any API docs.

File: docs/source/index.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Jax AI Stack
+=====================================
```
Comment (jakevdp) [inline docs/source/index.rst:2]: The length of the underline should match the length of the test (I believe sphinx generates warnings if it doesn't)

File: docs/source/index.html [hunk 3]
```diff
@@ -0,0 +1,105 @@
+<div class="doc-body">
+    <section class="hero">
+        <div>
+            <h1>Take Full Control of Your Data with JAX AI Stacks</h1>
```
Comment (jakevdp) [inline docs/source/index.html:4]: This looks like an early draft of the content copy – we'll need to update before we can merge this PR.

File: docs/source/conf.py (hunk 1 above)
Comment (barnesjoseph) [reply docs/source/conf.py]: Removed

File: docs/source/index.rst (hunk 2 above)
Comment (barnesjoseph) [reply docs/source/index.rst]: Updated

File: docs/source/index.html (hunk 3 above)
Comment (barnesjoseph) [reply docs/source/index.html:4]: It is an early version, I'm still waiting on final copy but will update soon.

**Comment by trallard** [comment]
@barnesjoseph we recently made updates to swap from the Alabaster theme to JupyterBook (as in the JAX docs) - ref: https://github.com/jax-ml/jax-ai-stack/pull/101

There might now be some conflicts between `main` and your PR. Would you be able to look into and resolve these? 
Feel free to ping if you need an extra set of eyes 👀

**Comment by barnesjoseph** [comment]
> @barnesjoseph we recently made updates to swap from the Alabaster theme to JupyterBook (as in the JAX docs) - ref: #101
> 
> There might now be some conflicts between `main` and your PR. Would you be able to look into and resolve these? Feel free to ping if you need an extra set of eyes 👀

I've merged main into my branch and resolved conflicts.

**Comment by trallard** [comment]
Thanks for the super quick update, @barnesjoseph! I just had a quick check, and everything seems okay with the PR build.

Something I noticed is that the tutorials/docs do not have a logo shown. For example, the [JAX documentation shows the JAX logo](https://jax.readthedocs.io/en/latest/beginner_guide.html#beginner-guide) (screenshots included)
Getting_Started_with_JAX_—_JAX_documentation

Tutorials_—_JAX_AI_Stack

So, I wanted to check if this is intentional or if the logo should also be added to the tutorial sidebar.

Also, when hovering over the buttons in `light` mode, the background colour darkens, which has decent enough contrast with the white text; however, in `dark` mode and due to the 

```
.button-primary:visited:hover {
    color: white !important;
}
```

The button background lightens, and the minimal contrast between this and the text makes it hard to read (which hinders accessibility). Thought I would flag this up but not mark as a hard fix.

**Comment by barnesjoseph** [comment]
> Thanks for the super quick update, @barnesjoseph! I just had a quick check, and everything seems okay with the PR build.
> 
> Something I noticed is that the tutorials/docs do not have a logo shown. For example, the [JAX documentation shows the JAX logo](https://jax.readthedocs.io/en/latest/beginner_guide.html#beginner-guide) (screenshots included) Getting_Started_with_JAX_—_JAX_documentation
> 
> Tutorials_—_JAX_AI_Stack
> 
> So, I wanted to check if this is intentional or if the logo should also be added to the tutorial sidebar.
> 
> Also, when hovering over the buttons in `light` mode, the background colour darkens, which has decent enough contrast with the white text; however, in `dark` mode and due to the
> 
> ```
> .button-primary:visited:hover {
>     color: white !important;
> }
> ```
> 
> The button background lightens, and the minimal contrast between it and the text makes it hard to read (which hinders accessibility). I thought I would flag this up but not mark it as a must-fix for now (screenshot attached below).
> 
> Jax_AI_Stack_—_JAX_AI_Stack

I've added some updated CSS to fix the button contrast issue on dark mode. Regarding the docs image comment, this is not in scope for the work my team was tasked with. Our work on this project has concluded as of a couple weeks ago and only pertains directly to the landing page, so any other changes from this point should be directed to Jake.

**Comment by trallard** [comment]
Thanks for the context. Will bring up the logo item with the rest of the team then.

**Pull request closed without merging**